server/gc/src/test/java/org/apache/accumulo/gc/GarbageCollectionTest.java

```diff
@@ -778,4 +838,152 @@ public void bulkImportReplicationRecordsPreventDeletion() throws Exception {
+    // if you add custom splits to the metadata at able boundaries, this can happen with a failed
+    // scan
+    // recall the ~tab:~pr for this first entry of a new table is empty, so there is now way to
+    // check the prior row. If you split a couple of tables in the metadata the table boundary
+    // , say table ids 2,3,4, and then miss scanning table 3 but get 4, it is possible other
+    // consistency checks will miss this
+    Set<TableId> tablesBefore = makeUnmodifiableSet("1", "2", "3", "4");
+    Set<TableId> tablesSeen = makeUnmodifiableSet("1", "2", "4");
+    Set<TableId> tablesAfter = makeUnmodifiableSet("1", "2", "3", "4");
+
+    GarbageCollectionAlgorithm gca = new GarbageCollectionAlgorithm();
+    String msg = assertThrows(RuntimeException.class,
+        () -> gca.ensureAllTablesChecked(tablesBefore, tablesSeen, tablesAfter)).getMessage();
+    assertTrue(msg.startsWith("Saw table IDs in ZK that were not in metadata table:"));
+
+  }
+
+  @Test
+  public void testZKHadNoTables() {
+    // this test simulates getting nothing from ZK for table ids, which should not happen,
+    // but just in case let's test
+    Set<TableId> tablesBefore = makeUnmodifiableSet();
+    Set<TableId> tablesSeen = makeUnmodifiableSet("1", "2");
+    Set<TableId> tablesAfter = makeUnmodifiableSet();
+
+    GarbageCollectionAlgorithm gca = new GarbageCollectionAlgorithm();
+    String msg = assertThrows(RuntimeException.class,
+        () -> gca.ensureAllTablesChecked(tablesBefore, tablesSeen, tablesAfter)).getMessage();
+    assertTrue(msg.startsWith("Saw no table ids in ZK but did see table ids in metadata table:"));
+  }
+
+  @Test
+  public void testMissingTableAndTableAdd() {
+    // simulates missing a table when checking references, and a table being added
+    Set<TableId> tablesBefore = makeUnmodifiableSet("1", "2", "3");
+    Set<TableId> tablesSeen = makeUnmodifiableSet("1", "2", "4");
+    Set<TableId> tablesAfter = makeUnmodifiableSet("1", "2", "3", "4");
+
+    GarbageCollectionAlgorithm gca = new GarbageCollectionAlgorithm();
+    String msg = assertThrows(RuntimeException.class,
+        () -> gca.ensureAllTablesChecked(tablesBefore, tablesSeen, tablesAfter)).getMessage();
+    assertTrue(msg.equals("Saw table IDs in ZK that were not in metadata table:  [3]"));
+  }
+
+  @Test
+  public void testMissingTableAndTableDeleted() {
+    // simulates missing a table when checking references, and a table being deleted
+    Set<TableId> tablesBefore = makeUnmodifiableSet("1", "2", "3", "4");
+    Set<TableId> tablesSeen = makeUnmodifiableSet("1", "2", "4");
+    Set<TableId> tablesAfter = makeUnmodifiableSet("1", "2", "3");
+
+    GarbageCollectionAlgorithm gca = new GarbageCollectionAlgorithm();
+    String msg = assertThrows(RuntimeException.class,
+        () -> gca.ensureAllTablesChecked(tablesBefore, tablesSeen, tablesAfter)).getMessage();
+    assertTrue(msg.equals("Saw table IDs in ZK that were not in metadata table:  [3]"));
```

Comment:
```suggestion
    assertEquals("Saw table IDs in ZK that were not in metadata table:  [3]", msg);
```
Can be simplified to assertEquals